app/Http/Controllers/Api/Instructor/AttendanceController.php

```diff
@@ -35,23 +35,18 @@ class AttendanceController extends Controller
      */
     public function store(StoreRequest $request): JsonResponse
     {
-        $instructorId = Auth::guard('instructor')->user()->id;
-
-        $courseIds = Course::where('instructor_id', $instructorId)
-            ->pluck('id')
-            ->toArray();
+        $course = Course::findOrFail($request->course_id);
 
-        if (! in_array($request->course_id, $courseIds, true)) {
-            // 講師の所有する講座でない場合はエラーを返す
-            throw new AuthorizationException('Forbidden, invalid instructor_id.');
-        }
+        // Policyによる認可チェック
+        $this->authorize('create', $course);
```

Comment:
明示的にAttendancePolicy::createとして呼び出すために次のように修正をお願いします。

```suggestion
        // Policyによる認可チェック
        $this->authorize('create', [Attendance::class, $course]);
```